runtime/auth/http-auth-aws/common/src/aws/smithy/kotlin/runtime/http/auth/AwsHttpSigner.kt

```diff
@@ -153,6 +154,15 @@ public class AwsHttpSigner(private val config: Config) : HttpSigner {
             }
         }
 
+        // Apply clock skew if necessary
+        signingConfig = if (attributes.contains(HttpOperationContext.ClockSkew)) {
+            val builder = signingConfig.toBuilder()
+            builder.signingDate = builder.signingDate?.plus(attributes[HttpOperationContext.ClockSkew])
```

Comment:
It's actually never null by this point, because when building the config [we either take the configured value or `Instant.now()`.](https://github.com/awslabs/smithy-kotlin/blob/bc522c642af169e7e34c128a818af79a38f8b799/runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/AwsSigningConfig.kt#L93)